internal/csi-addons/networkfence/fencing_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 The Ceph-CSI Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networkfence
+
+import (
+	"sort"
+	"testing"
+)
+
+func TestGetIPRange(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		cidr        string
+		expectedIPs []string
+	}{
+		{
+			cidr:        "10.23.56.67/32",
+			expectedIPs: []string{"10.23.56.67"},
+		},
+		{
+			cidr:        "62.76.47.12/30",
+			expectedIPs: []string{"62.76.47.12", "62.76.47.13", "62.76.47.14", "62.76.47.15"},
+		},
+		{
+			cidr:        "2001:4860:4860::8888/127",
+			expectedIPs: []string{"2001:4860:4860::8888", "2001:4860:4860::8889"},
+		},
+	}
+	for _, tt := range tests {
+		ts := tt
+		t.Run(ts.cidr, func(t *testing.T) {
+			t.Parallel()
+			got, err := getIPRange(ts.cidr)
+			if err != nil {
+				t.Errorf("error occurred in getting ip range")
+			}
+			sort.Strings(got)
+			for i := 0; i < len(ts.expectedIPs); i++ {
```

Comment:
this limits the check, there should also not be more IPs than that are expected

Reply:
Good catch! Updated now